gapil/compiler/plugins/encoder/encoder.go

```diff
@@ -523,22 +523,12 @@ func (e *encoder) encodeValue(s *compiler.S, ptr, buf *codegen.Value, ty semanti
 		return
 	case *semantic.Slice:
 		e.writeBlob(s, buf, func(buf *codegen.Value) {
-			root := s.LocalInit("root", ptr.Index(0, compiler.SliceRoot).Load().Cast(e.T.Size))
-			base := s.LocalInit("base", ptr.Index(0, compiler.SliceBase).Load().Cast(e.T.Size))
+			root := s.LocalInit("root", ptr.Index(0, compiler.SliceRoot).Load().Cast(e.T.Size)).Load()
```

Comment:
You don't need the `LocalInit` as that was there just to handle the branch of the if.

```
root := ptr.Index(0, compiler.SliceRoot).Load()
base := ptr.Index(0, compiler.SliceBase).Load()
size := ptr.Index(0, compiler.SliceSize).Load()
count := ptr.Index(0, compiler.SliceCount).Load()
pool := ptr.Index(0, compiler.SlicePool).Load()
```

I'll submit as-is, and I'll then make a follow up change.